sunkit_image/granule.py

```diff
@@ -0,0 +1,298 @@
+                elif mark:
+                    segmented_image_fixed[labeled_seg == value] = 0.5
+    return segmented_image_fixed
+
+
+def mark_brightpoint(segmented_image, data, resolution, bp_min_flux=None):
+    """
+    Mark brightpoints separately from granules - give them a value of 1.5.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect middles.
+    data : `numpy array`
+        The original image.
+    resolution : `float`
+        Spatial resolution (arcsec/pixel) of the data.
+    bp_min_flux : `float`, optional
+        Minimum flux per pixel for a region to be considered a brightpoint.
+        Default is `None` which will use data mean + 0.5 * sigma.
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndrray`
+        The segmented image with brightpoints marked as 1.5.
+    brightpoint_count: `int`
+        The number of brightpoints identified in the image.
+    granule_count: `int`
+        The number of granules identified, after re-classifcation of brightpoint.
+    """
+    bp_size_limit = 0.1  # Approximate max size of a photosphere bright point in square arcsec (see doi 10.3847/1538-4357/aab150)
+    bp_pix_upper_limit = bp_size_limit / (resolution**2)
+    bp_pix_lower_limit = 4 # Very small bright regions are likley artifacts
+    # General flux limit determined by visual inspection.
+    if bp_min_flux is None:
+        stand_devs = 0.5
+        bp_brightness_limit = np.mean(data) + stand_devs * np.std(data)
+    else:
+         bp_brightness_limit = bp_min_flux
+    if len(np.unique(segmented_image)) > 3:
+        raise ValueError("segmented_image must have only values of 1, 0 and a 0.5 (if dim centers marked)")
+    segmented_image_fixed = np.copy(segmented_image.astype(float)) # Make type float to enable adding float values
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    bp_count = 0
+    for value in values:
+        mask = np.zeros_like(segmented_image)
+        mask[labeled_seg == value] = 1
+        # Check that is a 1 (white) region.
+        if np.sum(np.multiply(mask, segmented_image)) > 0:
+            region_size = len(segmented_image_fixed[mask == 1])
+            tot_flux = np.sum(data[mask == 1])
+            # check that region is small.
+            if region_size < bp_pix_upper_limit:
+                # Check that region is not *too* small (likley an artifact)
+                if region_size > bp_pix_lower_limit:
+                    # Check that avg flux very high.
+                    if tot_flux / region_size > bp_brightness_limit:
+                        segmented_image_fixed[mask == 1] = 1.5
+                        bp_count += 1
+    gran_count = len(values) - 1 - bp_count  # Subtract 1 for IG region.
+    return segmented_image_fixed, bp_count, gran_count
+
+
+def kmeans_segment(data):

```

Comment:
This and `correlation` are for testing the original code if I recall?

If so, could these be moved into the test file and called there instead. After speaking to a few other people, if these are here, people will think they should call them.

Reply:
They are not used for testing the original code, per se, as much as providing a check on any segmentation performed. If the a segmented image is significantly off from what a simple kmeans segmentation would perform, that could indicate a problem (for example with the data).